### DISCUSSION TITLE
what is calculateAgeAdjustedRawValue() needed for?

### DISCUSSION THREAD

**Discussion by tynbendad** [NightscoutFoundation/xDrip#2137] [Q&A]
i see this in BgReading.java:

public void calculateAgeAdjustedRawValue(){
        final double adjust_for = AGE_ADJUSTMENT_TIME - time_since_sensor_started;
        if ((adjust_for > 0) && (!DexCollectionType.hasLibre())) {
            age_adjusted_raw_value = ((AGE_ADJUSTMENT_FACTOR * (adjust_for / AGE_ADJUSTMENT_TIME)) * raw_data) + raw_data;
            Log.i(TAG, "calculateAgeAdjustedRawValue: RAW VALUE ADJUSTMENT FROM:" + raw_data + " TO: " + age_adjusted_raw_value);
        } else {
            age_adjusted_raw_value = raw_data;
        }
    }

why are we scaling dexcom's raw_data by time since sensor started? this doesn't make sense to me - raw data is just that, raw. was this meant to fix the earlier floating point problem or for something else?

**Comment by Navid200** [comment]
Have you ever used G5 with xDrip in non-native mode?

**Reply by tynbendad** [reply]
We started on g5 a long long time ago and switched to g6 when it came out. Definitely used non native mode on it, aged raw wasn't around then and I didn't have a problem. Are you saying it's needed for g5? How do I add an official setting to turn it off?

**Reply by Navid200** [reply]
>  aged raw wasn't around then 

You can see it here, which is dated February 3, 2015, which was before there was a G6.
https://github.com/NightscoutFoundation/xDrip/blob/9b11f8a4fced8b785216feddfbda3c7ce9c16769/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java#L432

**Comment by tynbendad** [comment]
Ok, but there is at least one difference in whether the aged raw is used based on "calibration.check_in" between then and now. But, regardless of history I'm sure I don't want it on my g6 raw. I noticed today on a new sensor start during warmup that raw was about 300 but aged raw was 400, while actual BG was about 250. The g6 already bumps raw unfiltered on insertion, it doesn't need more bumping.

I am hoping to do raw vs native over time studies soon and think this will show up clearly in them, but that will take quite a while, can you help me come up with a setting switch or other acceptable change until then? I'm willing to make a PR if I'm pointed in the right direction.

**Reply by Navid200** [reply]
I have already told you if it was to me I would not have allowed raw capability to be added to G6.  It makes no sense and I see no reason to have it.  G6 provides factory calibration, which G5 or G4 did not.  Factory calibration is an evolutionary change.
There are individuals unfortunately, who insist on calibrating a G6.  Even the manufacturer tells you not to calibrate just for good measures:
https://www.dexcom.com/faqs/is-my-dexcom-sensor-accurate

There was a very important incentive for having non-native mode available for G5 and there was no con in that case as G5 did not have factory calibration.  But, if we use non-native mode for G6, we will be giving up factory calibration, and you are expressing having experienced the differences in different words.

I will help you assuming I can answer your questions.  But, it would be dishonest if I didn't tell you that I strongly disagree with what you are doing.

There are so many ways a user can set up xDrip to be misleading without realizing it.  I am one of those who does not like adding more settings that could cause that and many of my PRs attempt to remove such settings.

I prefer if you don't add a new setting unless absolutely necessary.
If you like to perform tests that you will be done with and will not need to do again, you can do it with a personal version of xDrip.

As I have told you before, I am not the person you need to convince to make changes to xDrip.

**Reply by tynbendad** [reply]
I have been using g6 since it was out and still do use raw, even with Anubis (with openaps and xdrip-js for the most part), I can use a personal version but since raw is a part of xdrip+ for g6 already it should be fixed for everyone so that it works properly. I guess I will do a PR to just remove this for g6, we can debate that in the PR.

**Comment by tynbendad** [comment]
pull request https://github.com/NightscoutFoundation/xDrip/pull/2140
I've added a chart showing data that I believe shows why aged raw doesn't make sense for G6.
